dask/bag/core.py

```diff
@@ -1526,10 +1526,14 @@ def groupby(
         if method is not None:
             raise Exception("The method= keyword has been moved to shuffle=")
         if shuffle is None:
-            shuffle = get_default_shuffle_method()
-            if shuffle == "p2p":
-                # Not implemented for Bags
+            try:
+                shuffle = get_default_shuffle_method()
+            except ImportError:
```

Comment:
`pyarrow` does not match `p2p` requirements, but we don't care.

Reply:
Wait, I'm confused. I wouldn't expect `get_default_shuffle_method` to ever raise

Reply:
See https://github.com/dask/distributed/pull/8157#issuecomment-1709648465

The idea is that we now require a pretty recent version of `pyarrow` so instead of silently falling back to `tasks` (or requiring it for all of `distributed`), we fail to alert users that have `pyarrow` installed but not the minimum version. Please defer to @fjetter to hash this out, he's taking over the PR while I'm out.

Reply:
The more troublesome issue is that `get_default_shuffle_method` gets shared between `dataframe` and `bag`.

Reply:
@jrbourbeau is this a problem for you? I want to avoid users accidentally falling back to tasks if they have an old pyarrow installed. This exception is raised early and they can either fix it by upgrading or by setting the config of shuffle to tasks.

Reply:
@jrbourbeau I will move forward with this now since you didn't reply in a little over a week. I'd like this to sit in main for at least a day before we release

Reply:
Sounds good if CI is happy